### PR TITLE
Phoneapp

### PR DESCRIPTION
**Pull request by maxeasy18**
Phoneapp was done with ES classes.
I have question about correct using my function _compareName and _compareStingValue
I don't understand hot to point from function _compareName  to _compareStingValue

### CODE REVIEW

File: phoneapp/src/app.js [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    _res += '-';
+    _res += number.slice(5,7);
+    _res += '-';
+    _res += number.slice(7,10);
+    return _res;
+  }
+
+
+
+  _prepareStringValue(name) {
```
Comment (OlegLustenko) [inline phoneapp/src/app.js]: prepare for what ?

Comment (OlegLustenko) [reply phoneapp/src/app.js]: Naming should include knowledge about domain layer and ideally be without details of implementation.

But I guess it's just a helper method which should not know anything about domain

File: phoneapp/src/app.js [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+      return this._compareName(user.name,name) || this._compareName(user.cname,name);
+    }); 
+
+    if( toDelUserIndex !== -1){
+      this.dataBase.splice(toDelUserIndex,1);
+    }  
+  }
+
+  findUser(name) {
+    return this.dataBase.reduce((acc,user) => {
```
Comment (OlegLustenko) [inline phoneapp/src/app.js]: please use .filter instead of .reduce

File: phoneapp/src/app.js [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+    const toEditUserIndex = this.dataBase.findIndex((user) => {
+      return user.id === id;
+    });
+    Object.keys(options).forEach((value) => { 
+      this.dataBase[toEditUserIndex][value] = options[value];    
+    })
+  }
+
+  filterBy(prop, value) {
+    return this.dataBase.reduce((acc,user) => {
```
Comment (OlegLustenko) [inline phoneapp/src/app.js]: please use .filter instead of .reduce

File: phoneapp/src/app.js [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+      }
+      if (propA > propB) {
+        return 1;
+      }
+      return 0;
+    })
+  }
+
+  editUser(id, options) {
+    const toEditUserIndex = this.dataBase.findIndex((user) => {
```
Comment (OlegLustenko) [inline phoneapp/src/app.js]: if you found that user probably you could update it at the same moment

are you remember such method ?
```js
const currentUser = {name: 'qwerty', surname: 'qwerrty'}
const newUserProperties = {name:'asdfg'};
const nextUser = {...curentUser, ...newUserProperties} // {name:'asdfg', surname:'qwerrty'}

```

File: phoneapp/src/app.js (hunk 3 above)
Comment (maxeasy18) [reply phoneapp/src/app.js]: Done

File: phoneapp/src/app.js (hunk 2 above)
Comment (maxeasy18) [reply phoneapp/src/app.js]: It is filter method too. By with to params.
Method changed

File: phoneapp/src/app.js [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+          {id:2, name:'dame2', cname:'Cname2', phone:'5222222222'},
+          {id:3, name:'hame3', cname:'qnbme3', phone:'3333333333'},
+          {id:4, name:'aame4', cname:'hname4', phone:'744224444'},
+      ]    
+  }
+
+  transformNumber(number) {
+    let _res = '(';
+    let _number = String(number);
+    _res += number.slice(0,3);
```
[the remark this answers is not in the document]
Comment (maxeasy18) [reply phoneapp/src/app.js]: Done

File: phoneapp/src/app.js (hunk 4 above)
Comment (maxeasy18) [reply phoneapp/src/app.js]: No, I don't remeber.
But I've tried and it works.

**Comment by maxeasy18** [comment]
Changes applied

**Pull request merged**